### ISSUE TITLE
Green screen glitch

### ISSUE THREAD

**Issue by Arshimonde** [boonya/rtsp-video-recorder#35]
Hello brother boonya, thanks a lot for this great library.

I found a bug in the ffmpeg command that makes one of our RTSP camera stream saved as a video mp4 but in a green screen glitch (example image bellow).

Image: 
![](https://i.ytimg.com/vi/hJK8X8H29Es/maxresdefault.jpg)

Solution I have done to resolve the issue:
in the file `recorder.ts` in function `spawnFFMPEG` I removed the parameter `-vcodec copy`

Notes: 
- VLC can read it properly but other video players cannot.
- Even after removing the parameter  `-vcodec copy` from ffmpeg command everything works perfectly.

**Comment by boonya** [comment]
Hello brother @Arshimonde, thank you for such pleasant words )

Since I have bought several different IP Web Cams I saw this issue as well. But I didn't have enough time to investigate it. I will try and investigate you solution soon and if it works I will apply it to the source code.

Thanks again

**Comment by boonya** [comment]
Okay, @Arshimonde , I investigated a lot and now I see that issue is not because of codec option. This is most likely because of network interface bandwidth. My investigation still in progress but you may try to decrease bitrate of your camera stream.
And let me know please.

**Comment by Arshimonde** [comment]
Hello brother @boonya, sorry for the delay. hope you're doing great.

I uncommented `-vcodec copy` parameters and tested again today, with lowering the bitrate to` 512kbps`, and with the original bitrate `1792 kbps`.

**Note:** 
I have 4 _analog_ cameras connected to _Truvision DVR_, only the first camera which has this issue.

**Result of test**:
it works on both bitrates.

**Remark:** 
Because VLC read the video with green screen glitch, I think it is indeed a format (coding / decoding) issue 
but now it works even with those parameters with QuickTime player (last time wasn't).

**Comment by boonya** [comment]
`-vcodec copy` means that a stream won't be transcoded in any other encodings. So it will be piped to the mp4 container as is. As far as got it, if you remove this option at all `ffmpeg` applies some codec automatically.

I created this lib to be working on RaspberryPI, as far this board is pretty silly it can not transcode video streams, so I applied `-vcodec copy`.

In next version I am going to implement ability to pass custom parameters to `ffmpeg` directly. It will allow you to tune it up as you wish.
